test/EndToEndTests/Tests/Client/Build.Desktop/ClientTests/ClientQueryTests.cs

```diff
@@ -175,6 +175,8 @@ public void QueryEntityNavigationWithImplicitKeys()
                     Assert.IsTrue(testCase.Value);
                     Assert.AreEqual(400, ex.Response.StatusCode);
                     StringResourceUtil.VerifyDataServicesString(ClientExceptionUtil.ExtractServerErrorMessage(ex), "BadRequest_KeyCountMismatch", "Microsoft.Test.OData.Services.AstoriaDefaultService.Message");
+                    //InnerException for DataServiceClientException must be set with the exception response frpom the server.
+                    Assert.IsTrue(ex.InnerException.InnerException != null, "InnerException for DataServiceClientException has not been set.");

```

Comment:
Maybe use `Assert.NotNull` or equivalent if available?

Reply:
Is it possible to also test whether the `ex.InnerException.InnerException` contains the expected information?

Reply:
> or equivalent if availab

Serves the same purpose.

Reply:
> Is it possible to also test whether the ex.InnerException.InnerException contains the expected information?

Good point. In this case we will check for a well formed `ODataErrorException ` Object. content is not necessary.